spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -81,147 +71,586 @@ public class StreamApplicationTests {
-`spring.cloud.stream.bindings.default.destination=foo` and `spring.cloud.stream.bindings.tap.destination=bar`,
-and the result is 2 bindings to an external broker with destinations called "foo" and "bar".
 
-==== Inter-app Communication
+=== Application structure
 
-While Spring Cloud Stream makes it easy for individual boot apps to connect to messaging systems, the typical scenario
-for Spring Cloud Stream is the creation of multi-app pipelines, where microservice apps are sending data to each other.
-This can be achieved by correlating the input and output destinations of adjacent apps, as in the following example.
+A Spring Cloud Stream application consists of a middleware-neutral core that uses a Spring Messaging-based API and Spring Integration components, and communicates with the outside world through input and output channels that are managed and injected into it by the framework. A Spring Cloud Stream abstraction named a binder provides the bridging between the channels and the external brokers. Different binder implementations exist for different types of middleware, such as Kafka, Rabbit MQ, Redis or Gemfire, and an extensible API allows you to write your own binder.
 
-Supposing that the design calls for the `time-source` app to send data to the `log-sink` app, we will use a
-common destination named `ticktock` for bindings within both apps. `time-source` will set
-`spring.cloud.stream.bindings.output.destination=ticktock`, and `log-sink` will set
-`spring.cloud.stream.bindings.input.destination=ticktock`.
+.Spring Cloud Stream Application
+image::SCSt-with-binder.png[width=300,scaledwidth="50%"]
 
-==== Consumer Group Support
+Spring Cloud Stream uses Spring Boot for configuration, and the binder abstraction makes it possible for Spring Cloud Stream applications to be flexible in terms of how it connects to the middleware.
 
-Spring Cloud Stream is a library focusing on building message-driven microservices, and more specifically stream
-processing applications. In such scenarios, communication between different logical applications follows a
-publish-subscribe pattern, with data being broadcast through a shared topic, but at the same time, it is important to
-be able to scale up by creating multiple instances of a given application, which are in a competing consumer
-relationship with each other.
+This can mean, for example, dynamically choosing the destinations that these channels connect to at runtime (e.g. Kafka topics or Rabbit MQ exchanges), via external configuration properties in any form that is supported by Spring Boot (application arguments, environment variables, `application.yml` files, etc). Taking the source example from the previous section, providing the `spring.cloud.stream.bindings.output.destination=raw-sensor-data` property to the application will cause it to write to the `raw-sensor-data` Kafka topic, or the `orders` exchange in Rabbit.
```

Comment:
> or the `orders` exchange in Rabbit

I do not see any reference of `orders` exchange in the previous section. Perhaps, this can be called out simply as "topic exchange in RabbitMQ"?